src/magic.rs

```diff
@@ -39,7 +39,15 @@ pub fn detect_mime_type(buffer: &[u8]) -> Result<Mime, Box<dyn std::error::Error
         .stdin
         .as_mut()
         .expect("Forgot to pipe stdin?")
-        .write_all(buffer)?;
+        // extract only the first 4kb of interesting bytes
+        // that's sufficient for detecting the mime type,
+        // otherwise the method `write_all` fails with
+        // "Broken pipe (os error 32)" error if the data
+        // being piped is large enough to be terminated
+        // due to signal SIGPIPE as the file command
+        // reads maximum 1048576 bytes from the data.
```

Comment:
Assuming there's a reason behind this limit shouldn't we rather truncate at this amount of bytes then at `4096`?

Reply:
I think I will change it to that value. I chose the 4096 bytes as it’s sufficient for detecting the image bytes (the possible smallest valid images are far less than 4096 bytes)and it’s more faster. Your choice. 🙂